account_setup.py

```diff
@@ -0,0 +1,348 @@
+                {
+                    'role_definition_id': contributor_role.id,
+                    'principal_id': principal_id
+                }
+            )
+            break
+        except CloudError as e:
+            # ignore error if service principal has not yet been created
+            time.sleep(1)
+            if i == 10:
+                raise e
+
+
+def format_secrets(**kwargs):
+    '''
+    Retuns the secrets for the created resources to be placed in secrets.yaml
+    The following form is returned:
+
+        service_principal:
+            tenant_id: <AAD Diretory ID>
+            client_id: <AAD App Application ID>
+            credential: <AAD App Password>
+            batch_account_resource_id: </batch/account/resource/id>
+            storage_account_resource_id: </storage/account/resource/id>
+    '''
+    return yaml.dump({"service_principal": kwargs}, default_flow_style=False)
+
+
+def prompt_for_confirmation(prompt, deny_error, unrecognized_input_error):
+    """
+        Prompt user for confirmation, 'y' for confirm, 'n' for deny
+        :param prompt: str
+        :param deny_error: Exception
+        :param unrecognized_input_error: Exception
+        :return None if prompt successful, else raises error
+    """
+    confirmation = input(prompt).lower()
+    for i in range(3):
+        if confirmation == "n":
+            raise deny_error
+        elif confirmation == "y":
+            break
+        elif confirmation != "y" and i == 2:
+            raise unrecognized_input_error
+        confirmation = input("Please input 'y' or 'n': ").lower()
+
+
+def prompt_with_default(key, value):
+    user_value = input("{0} [{1}]: ".format(key, value))
+    if user_value != "":
+        return user_value
+    else:
+        return value
+
+
+if __name__ == "__main__":
+    print("\nGetting credentials.")
+    # get credentials and tenant_id
+    creds, subscription_id = credentials.get_azure_cli_credentials()
+    subscription_client = SubscriptionClient(creds)
+    tenant_ids = [tenant.id for tenant in subscription_client.tenants.list()]
+    if len(tenant_ids) != 1:
+        raise AccountSetupError("More than one tenant configured on your subscription.")
+    else:
+        tenant_id = tenant_ids[0]
+
+    print("Input the desired names and values for your Azure resources. "\
+          "Default values are provided in the brackets. "\
+          "Hit enter to use default.")
+    kwargs = {
+        "region": prompt_with_default("Azure Region", DefaultSettings.region),
+        "resource_group": prompt_with_default("Resource Group Name", DefaultSettings.resource_group),
+        "storage_account": prompt_with_default("Storage Account Name", DefaultSettings.storage_account),
+        "batch_account": prompt_with_default("Batch Account Name", DefaultSettings.batch_account),
+        # "virtual_network_name": prompt_with_default("Virtual Network Name", DefaultSettings.virtual_network_name),
```

Comment:
un-comment?

Reply:
maybe this could be a 'leave blank for no vnet'

Reply:
I left this commented since we don't (currently) have a way to output the subnet_id since it is in cluster.yaml and everything else is in secrets.yaml.